### PR TITLE
 APIs for distro specific VM network configurations

### PR DESCRIPTION
**Pull request by balamuruhans**
Network testcases require distro specific network configuration files
and the parameters done during tests.

Signed-off-by: Balamuruhan S <bala24@linux.vnet.ibm.com>
Signed-off-by: Sudeesh John <sudeesh@linux.vnet.ibm.com>

### CODE REVIEW

File: virttest/libvirt_vm.py [hunk 1]
```diff
@@ -2456,6 +2456,56 @@ def get_interface_mac(self, interface):
         session.close()
         return mac.strip()
 
+    def get_network_cfg_file(self, iface_name):
+        """
+        Get absolute network cfg file path of the vm instance.
+
+        :param iface_name: Name of the network interface
```
Comment (will-Do) [inline virttest/libvirt_vm.py]: better to add ':return'

File: virttest/libvirt_vm.py [hunk 2]
```diff
@@ -2456,6 +2456,56 @@ def get_interface_mac(self, interface):
...
+                          net_mask, **dargs):
+        """
+        Set network cfg parameter of the vm instance and returns the list
+
+        :param iface_name: Name of the network interface
+        :param mac_addr: Mac address of the network interface
+        :param boot_proto: static ip or dhcp of the interface
+        :param ip_addr: ip address for the interface
+        :param net_mask: network mask for the interface
+        :param dargs: dict of additional attributes
```
Comment (will-Do) [inline virttest/libvirt_vm.py]: And here

Comment (will-Do) [inline virttest/libvirt_vm.py]: Just curious about the usage of this function:
1. What's the purpose about return a list?
2. From the function name 'set_net_param', it's hard to understand what it does

**Comment by bssrikanth** [comment]
@balamuruhans  why not have this function part of utils_net instead of libvirt_vm.py... since in qemu as well we need this function.. can we make it common?

**Comment by balamuruhans** [comment]
@bssrikanth Thanks 👍 I agree with your suggestion, I am working on comments given by @will-Do. Once done and will place these APIs in utils_net.py and test.

File: virttest/libvirt_vm.py (hunk 2 above)
Comment (balamuruhans) [inline virttest/libvirt_vm.py]: @will-Do this API can used to configure and test guest with different n/w configurations and to persist after reboot, destroy and boot, migration etc.,
1. List can be dumped to network script file with \n as linesep
2. sure, I have changed the name as `create_vm_network_script`

File: virttest/libvirt_vm.py (hunk 1 above)
Comment (balamuruhans) [inline virttest/libvirt_vm.py]: @will-Do done.

**Comment by balamuruhans** [comment]
@bssrikanth done.

**Comment by bssrikanth** [comment]
Thanks Bala.

**Comment by balamuruhans** [comment]
@xutian @will-Do Please help to review this PR

**Comment by xutian** [comment]
Looks good to me.

**Comment by balamuruhans** [comment]
@xutian, Thank you

**Comment by xutian** [comment]
@will-Do any comments for this PR,  if no, let me merge this PR.

Thanks,
Xu

**Comment by balamuruhans** [comment]
@xutian Thank you

**Pull request merged**